tests/test_lib_events.py

```diff
@@ -2867,4 +2867,86 @@ def test_20_sendmail_googleurl(self):
+                            "role": "admin",
+                            "realm": ""}
+        g.audit_object = audit_object
+
+        builder = EnvironBuilder(method='POST',
+                                 data={'serial': "OATH123456"},
+                                 headers={})
+
+        env = builder.get_environ()
+        # Set the remote address so that we can filter for it
+        env["REMOTE_ADDR"] = "10.0.0.1"
+        g.client_ip = env["REMOTE_ADDR"]
+        req = Request(env)
+        req.all_data = {"serial": "OATH123456",
+                        "user": "cornelius"}
+        req.User = User("cornelius", self.realm1)
+        resp = Response()
+        resp.data = """{"result": {"value": true}}"""
+        options = {"g": g,
+                   "request": req,
+                   "response": resp,
+                   "handler_def": {"options":
+                                       {"filename": "test{serial}.txt",
+                                        "body": "{serial}, {user}"}
+                                   }
+                   }
+
+        un_handler = UserNotificationEventHandler()
+        res = un_handler.do("savefile", options=options)
+        self.assertTrue(res)
+        # check, if the file was written with the correct contents
+        with open("tests/testdata/testOATH123456.txt") as f:
+            l = f.read()
+        self.assertEqual(l, "OATH123456, Cornelius")
+        os.remove("tests/testdata/testOATH123456.txt")
+
+        # Check what happens if we try to write outside of spooldir
+        options = {"g": g,
+                   "request": req,
+                   "response": resp,
+                   "handler_def": {"options":
+                                       {"filename": "../../../test{serial}.txt",
+                                        "body": "{serial}, {user}"}
+                                   }
+                   }
+
+        un_handler = UserNotificationEventHandler()
+        un_handler.do("savefile", options=options)
+        # Check the log file for an error. There might be faster possibilities but more robust
+        with open("privacyidea.log") as f:
+            lines = f.read().splitlines()
+            last_line = lines[-1]
+        self.assertIn("Cannot write outside of spooldir tests/testdata/!", last_line)
+
+        # Check what happens if the file can not be written
+        options = {"g": g,
+                   "request": req,
+                   "response": resp,
+                   "handler_def": {"options":
+                                       {"filename": "test{serial}.txt",
+                                        "body": "{serial}, {user}"}
+                                   }
+                   }
+
+        # create a file, that is not writable
+        with open("tests/testdata/testOATH123456.txt", "w") as f:
+            f.write("empty")
+        os.chmod("tests/testdata/testOATH123456.txt", 0x400)
```

Comment:
``0x400`` is hexadecimal, but I think file modes are octal? should probably be ``0o400``

Reply:
done in 2b85a2fe